components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt

```diff
@@ -510,17 +511,18 @@ abstract class AbstractFetchDownloadService : Service() {
     @TargetApi(Build.VERSION_CODES.P)
     @Suppress("Deprecation")
     private fun useFileStreamLegacy(download: DownloadState, append: Boolean, block: (OutputStream) -> Unit) {
+        val fileName = download.fileName ?: throw IllegalStateException("A fileName for a download is required")
```

Comment:
Should we be throwing an exception currentDownloadJobState as well insetad of the return?

Reply:
That's good question, if we do that will be will crashing the app but we will be discovering an illegal state in which the app shouldn't be, we should weight that 👍

Reply:
We could keep as it is to collect more information and see it the crash it related, what do you think?

Reply:
That sounds good to me! Let's keep as-is.

Reply:
👍